activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -657,6 +670,11 @@ def arel_from_relation(relation)
             relation
           end
         end
+
+        def extract_table_ref_from_insert_sql(sql)
+          sql[/into\s("[A-Za-z0-9_."\[\]\s]+"|[A-Za-z0-9_."\[\]]+)\s*/im]
+          $1.strip if $1
```

Comment:
This is weird, why not the more obvious:

```suggestion
          if sql =~ /into\s("[A-Za-z0-9_."\[\]\s]+"|[A-Za-z0-9_."\[\]]+)\s*/im
            $1.strip
          end
```

?